src/main/java/org/traccar/protocol/Gl200TextProtocolDecoder.java

```diff
@@ -42,11 +42,45 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.TimeZone;
+import java.util.HashMap;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class Gl200TextProtocolDecoder extends BaseProtocolDecoder {
 
+    private static final HashMap<String, String> DEVICE_MODELS = new HashMap<String, String>() {{
+        put("02", "GL200");
+        put("04", "GV200");
+        put("06", "GV300");
+        put("08", "GMT100");
+        put("09", "GV50P");
+        put("0F", "GV55");
+        put("10", "GV55 LITE");
+        put("11", "GL500");
+        put("1A", "GL300");
+        put("1F", "GV500");
+        put("25", "GV300");
+        put("27", "GV300W");
+        put("2C", "GL300W");
+        put("2F", "GV55");
+        put("30", "GL300");
+        put("35", "GV200");
+        put("36", "GV500");
+        put("3F", "GMT100");
+        put("41", "GV75W");
+        put("50", "GV55W");
+        put("52", "GL50");
+        put("55", "GL50B");
+        put("5E", "GV500MAP");
+        put("6E", "GV310LAU");
+        put("C2", "GV600M");
+        put("F1", "GV350M");
+        put("F8", "GV800W");
+        put("FC", "GV600W");
+        put("802004", "GV58LAU");
+        put("802005", "GV355CEU");
+    }};
```

Comment:
Why are there two brackets?

Reply:
Double brace initialisation so we can use the initialiser block